testing/config.go

```diff
@@ -0,0 +1,62 @@
+package ibctesting
+
+import (
+	"time"
+
+	connectiontypes "github.com/cosmos/ibc-go/modules/core/03-connection/types"
+	channeltypes "github.com/cosmos/ibc-go/modules/core/04-channel/types"
+	"github.com/cosmos/ibc-go/modules/core/exported"
+	ibctmtypes "github.com/cosmos/ibc-go/modules/light-clients/07-tendermint/types"
+)
+
+type ClientConfig interface {
+	GetClientType() string
+}
+
+type TendermintConfig struct {
+	TrustLevel                   ibctmtypes.Fraction
+	TrustingPeriod               time.Duration
+	UnbondingPeriod              time.Duration
+	MaxClockDrift                time.Duration
+	AllowUpdateAfterExpiry       bool
+	AllowUpdateAfterMisbehaviour bool
+}
+
+func NewTendermintConfig() *TendermintConfig {
+	return &TendermintConfig{
+		TrustLevel:                   DefaultTrustLevel,
+		TrustingPeriod:               TrustingPeriod,
+		UnbondingPeriod:              UnbondingPeriod,
+		MaxClockDrift:                MaxClockDrift,
+		AllowUpdateAfterExpiry:       false,
+		AllowUpdateAfterMisbehaviour: false,
+	}
+}
+
+func (tmcfg *TendermintConfig) GetClientType() string {
+	return exported.Tendermint
+}
+
+type ConnectionConfig struct {
+	DelayPeriod uint64
+	Version     *connectiontypes.Version
+}
+
+func NewConnectionConfig() *ConnectionConfig {
+	return &ConnectionConfig{
+		DelayPeriod: DefaultDelayPeriod,
+		Version:     ConnectionVersion,
+	}
+}
+
+type ChannelConfig struct {
+	Version string
+	Order   channeltypes.Order
+}
+
+func NewChannelConfig() *ChannelConfig {
```

Comment:
Channel Config should also contain a successful ACK field and an unsuccessful ACK field, so app developers can specify to the testing package what their acks look like.

Testing package can just write these acks as needed to test app functionality

Reply:
It is possible acknowledgements might contain specific information. I'd prefer to leave this flexible for now rather than only supporting two types of acknowledgements